source/proximity_list/proximity_list.cpp

```diff
@@ -169,7 +158,32 @@ auto ProximityList<T>::find_within_impl(const GeographicalCoord& coord,
     int nb_found = radius_search(NN_index, coord, radius, size, indices, distances);
     assert(indices.size() == 1);
     assert(distances.size() == 1);
-    return make_result<T>(coord, items, indices[0], distances[0], nb_found, IndexCoord{});
+
+    std::vector<typename ReturnTypeTrait<T, IndexCoord>::ValueType> res;
+    auto op = [](const Item& item, float /*unused*/) { return std::make_pair(item.element, item.coord); };
+    make_result<T>(coord, items, indices[0], distances[0], nb_found, res, op);
+
+    return res;
+}
+
+template <class T>
+auto ProximityList<T>::find_within_impl(const GeographicalCoord& coord,
+                                        double radius,
+                                        int size,
+                                        IndexCoordDistance /*unused*/) const
+    -> std::vector<typename ReturnTypeTrait<T, IndexCoordDistance>::ValueType> {
```

Comment:
This function is the same as`find_within_impl` above. 
The same way as previously, if you use `unused` parameter as template parameter, you should be able to remove the version of `find_within_impl` above.

Reply:
no, they're not the same... it's actually at dispatch tag for function overload... they're used as "typename" not as "object"

Reply:
i've tried to refactorize these two functions, since you mentioned they're very much alike, it just complicates the codes more.... I've no magic solution this time :/